### PR TITLE
Import https://developer.ubuntu.com/snappy/guides/mir-snaps/

### PR DESCRIPTION
**Pull request by dholbach**
LP: #1541404

### CODE REVIEW

File: docs/mir-snaps.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+Quickly double-check Mir server is running, if not just start the service.
+```
+ubuntu# sudo snappy install mir-client*.snap --allow-unauthenticated
+```
+
+At this point you should see some Qml clocks.
+
+*Note: 2016-01-26*
+ - This snap is currently unconfined. Working with the security team to rely
+   on the mir-client capability of the system.
```
Comment (sergiusens) [inline docs/mir-snaps.md]: Maybe use quotation here? `>`

```markdown
> *Note: 2016-01-16*
> - This snap is currently unconfined. Working with the security team to rely
>   on the mir-client capability of the system.
```

so it looks like:

> *Note: 2016-01-16*
> - This snap is currently unconfined. Working with the security team to rely
>   on the mir-client capability of the system.

File: docs/mir-snaps.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+applications supported then Ubuntu Personal is the answer.
+
+This document targets Xenial Ubuntu Core and assumes your host machine is
+running Xenial (up to date) and hence using Snapcraft 2.0.
+
+*NOTE: 2016-02-03*
+- snapcraft 2.1 came out today and requires changes. Will followup with
+  updates.
+- also, the most recent Xenial kernel has [a bug creating issues for
+  mir](https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1540731)
```
Comment (sergiusens) [inline docs/mir-snaps.md]: same quotation comment (read below)

**Comment by dholbach** [comment]
@sergiusens - how would you like me to import the images from http://developer.ubuntu.com/snappy/guides/mir-snaps/?

File: docs/mir-snaps.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+second snap is strictly an example of a `mir-client-app`, which developers can
+use as a template to create their own. If you are experimenting and simply
+building both snaps at the same time you should experience no problems,
+however once the `mir-server` snap is in the store, it is important that the
+mir-client use the `libmirclient` version corresponding to the `libmirserver`
+within their snap.
+
+The `mir-snap` is only targeted for Ubuntu Core. It is intended for
+kiosk-like products and applications. Products requiring many separate UI
+applications supported then Ubuntu Personal is the answer.
```
Comment (kyrofa) [inline docs/mir-snaps.md]: Consider a slight rephrase here: "If products require many separate UI applications, then they need Ubuntu Personal."

File: docs/mir-snaps.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+### Get the Mir Snap running first
+
+If you haven’t already, install the snapcraft tools.
+```
+$ sudo add-apt-repository ppa:snappy-dev/tools
+$ sudo apt-get update
+$ sudo apt-get upgrade
+$ sudo apt-get install snapcraft
```
Comment (kyrofa) [inline docs/mir-snaps.md]: Might be worth pointing back to get-started.md rather than duplicating the instructions here.

**Comment by kyrofa** [comment]
We can always upload images in an "images" or "docs" branch, and link to them from here.

File: docs/mir-snaps.md (hunk 4 above)
Comment (sergiusens) [inline docs/mir-snaps.md]: +1

Comment (sergiusens) [inline docs/mir-snaps.md]: Also, the ppa is no longer needed

**Comment by dholbach** [comment]
@kyrofa - @sergiusens - just let me know if the docs branch is what you want to go ahead with or if we do it like in https://github.com/ubuntu-core/snapcraft/blame/master/README.md#L33

**Comment by dholbach** [comment]
Ok, branch is updated now.

File: docs/mir-snaps.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+from the Ubuntu Software Center). Select the icon for "New Virtual Machine"
+(or menu to "File", "New Virtual Machine"). In the dialog select the radio
+button for "Import Existing Disk Image", browse to your `*.img` and select
+it. You can leave the other defaults. Once you select your way forward, it
+should launch another window with your Snappy VM in it. Once the system
+settles it will provide you a prompt. Just click on the window to interact
+with the VM, to escape select "control+alt" keys. To login, the Snappy image
+user is "ubuntu" and password is "ubuntu". Check your ip with ipconfig.
+
+[vm-image]: https://raw.githubusercontent.com/ubuntu-core/snapcraft/master/docs/images/ubuntucore_in_vmm.png
```
Comment (kyrofa) [inline docs/mir-snaps.md]: Doesn't this require a !? Referencing the example here: https://guides.github.com/features/mastering-markdown/

**Comment by dholbach** [comment]
Seems to work: https://github.com/dholbach/snapcraft/blob/1541404/docs/mir-snaps.md

File: docs/mir-snaps.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+If you run out of memory from loading too many snaps, you can also grow your
+image size. For example:
+```
+$ qemu-img resize xenial_core_amd64.img +1G
+```
+
+[vm-image]: https://rawgit.com/dholbach/snapcraft/1541404/docs/images/ubuntucore_in_vmm.png
+[core-image]: https://rawgit.com/dholbach/snapcraft/1541404/docs/images/core_running_mir.png
+[clock-image]: https://rawgit.com/dholbach/snapcraft/1541404/docs/images/clocks_on_mir_on_ubuntucore.png
```
Comment (kyrofa) [inline docs/mir-snaps.md]: Looks good! Alright, go ahead and update these and I'm :+1: .

**Comment by kyrofa** [comment]
Looks good to me :+1: . I'll let @sergiusens take one more look before merge.

**Comment by sergiusens** [comment]
:+1: Next time though, don't drop the coverage by 0% :wink:

**Pull request merged**